File: Top-Interview-Questions-Leetcode/Easy-Collections/strings/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq = [0 for _ in range(26)]
        
        for i in range(len(s)):
            char = ord(s[i]) - ord('a')
            freq[char] += 1
        
        for i in range(len(s)):
            char = ord(s[i]) - ord('a')
            
            if freq[char] == 1:
                return i
        
        return -1

# Example usage
def main():
    s = "phanthanhdang"
    solution = Solution()
    result = solution.firstUniqChar(s)
    print(result)

if __name__ == "__main__":
    main()

File: Top-Interview-Questions-Leetcode/Easy-Collections/strings/first-unique-character-in-a-string.cpp
#include <iostream>
#include <vector>
#include <string>

class Solution {
public:
    int firstUniqChar(std::string s) {
        std::vector<int> freq(26);
        
        for (char ch : s) freq[ch - 'a']++;
        
        for (int i = 0; i < s.size(); ++i) {
            if (freq[s[i] - 'a'] == 1) return i;
        }
        
        return -1;
    }
};

// Example usage
int main() {
    Solution solution;
    std::string s = "phanthanhdang";
    int result = solution.firstUniqChar(s);
    std::cout << "The index of the first unique character is: " << result << std::endl;
    return 0;
}